test/built-ins/Temporal/Calendar/prototype/monthDayFromFields/overflow-constrain.js

```diff
@@ -10,82 +10,56 @@ info: |
   3. Assert: calendar.[[Identifier]] is "iso8601".
   4. If Type(fields) is not Object, throw a TypeError exception.
   5. Set options to ? GetOptionsObject(options).
-  6. Let result be ? ISOMonthDayFromFields(fields, options).
-  7. Return ? CreateTemporalMonthDay(result.[[Month]], result.[[Day]], calendar, result.[[ReferenceISOYear]]).
+  6. Set fields to ? PrepareTemporalFields(fields, « "day", "month", "monthCode", "year" », « "day" »).
+  7. Let overflow be ? ToTemporalOverflow(options).
+  8. Perform ? ISOResolveMonth(fields).
+  9. Let result be ? ISOMonthDayFromFields(fields, overflow).
+  10. Return ? CreateTemporalMonthDay(result.[[Month]], result.[[Day]], "iso8601", result.[[ReferenceISOYear]]).
 includes: [temporalHelpers.js]
 features: [Temporal]
 ---*/
 
 const cal = new Temporal.Calendar("iso8601");
 const opt = { overflow: "constrain" };
 
-let result = cal.monthDayFromFields({ year: 2021, month: 1, day: 133 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M01", 31, "day is constrained to 31 in month 1");
-result = cal.monthDayFromFields({ year: 2021, month: 2, day: 133 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M02", 28, "day is constrained to 28 in month 2 (year 2021)");
-result = cal.monthDayFromFields({ year: 2021, month: 3, day: 9033 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M03", 31, "day is constrained to 31 in month 3");
-result = cal.monthDayFromFields({ year: 2021, month: 4, day: 50 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M04", 30, "day is constrained to 30 in month 4");
-result = cal.monthDayFromFields({ year: 2021, month: 5, day: 77 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M05", 31, "day is constrained to 31 in month 5");
-result = cal.monthDayFromFields({ year: 2021, month: 6, day: 33 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M06", 30, "day is constrained to 30 in month 6");
-result = cal.monthDayFromFields({ year: 2021, month: 7, day: 33 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M07", 31, "day is constrained to 31 in month 7");
-result = cal.monthDayFromFields({ year: 2021, month: 8, day: 300 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M08", 31, "day is constrained to 31 in month 8");
-result = cal.monthDayFromFields({ year: 2021, month: 9, day: 400 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M09", 30, "day is constrained to 30 in month 9");
-result = cal.monthDayFromFields({ year: 2021, month: 10, day: 400 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M10", 31, "day is constrained to 31 in month 10");
-result = cal.monthDayFromFields({ year: 2021, month: 11, day: 400 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M11", 30, "day is constrained to 30 in month 11");
-result = cal.monthDayFromFields({ year: 2021, month: 12, day: 500 }, opt);
-TemporalHelpers.assertPlainMonthDay(result, "M12", 31, "day is constrained to 31 in month 12");
```

Comment:
These have been consolidated into the second `forEach`, which covers the smallest affected day for each month along with an excessive value.